supabase/async_client.py

```diff
@@ -0,0 +1,186 @@
+import re
+from typing import Any, Coroutine, Dict, Union
+
+from httpx import Timeout
+from postgrest import (
+    AsyncFilterRequestBuilder,
+    AsyncPostgrestClient,
+    AsyncRequestBuilder,
+)
+from postgrest.constants import DEFAULT_POSTGREST_CLIENT_TIMEOUT
+from supafunc import FunctionsClient
+
+from .exceptions import SupabaseException
+from .lib.auth_client import SupabaseAuthClient
+from .lib.client_options import ClientOptions
+from .lib.storage_client import SupabaseStorageClient
+
+
+class AsyncSupabaseClient:
```

Comment:
Am wondering if we should do away with the sync binding and generate it via `unasync` like how it's done in `Postgrest-py` and `gotrue-py` I think @anand2312 has more expertise here so will let him chime in.

Reply:
Yeah we'd definitely want to generate the sync client with unasync, but I don't think `realtime` is at a useable enough state for us to guarantee a sync interface which would make it difficult to generate two clients here (imo realtime should only have an async interface so only an AsyncSupabaseClient should have realtime features but that's upto debate)

Reply:
If this PR is approved, I’ll be adding realtime to the async class. This was initially my intention.